bitnami/kafka/templates/statefulset.yaml

```diff
@@ -96,18 +96,17 @@ spec:
           args:
             - -ec
             - |
-              mkdir -p /bitnami/kafka
-              chown -R "{{ .Values.podSecurityContext.runAsUser }}:{{ .Values.podSecurityContext.fsGroup }}" "/bitnami/kafka"
-              mkdir -p /opt/bitnami/kafka/logs
-              chown -R "{{ .Values.podSecurityContext.runAsUser }}:{{ .Values.podSecurityContext.fsGroup }}" "/opt/bitnami/kafka/logs"
-          securityContext:
-            runAsUser: 0
+              chown -R "{{ .Values.podSecurityContext.runAsUser }}:{{ .Values.podSecurityContext.fsGroup }}" "{{ .Values.persistence.mountPath }}"
+              chown -R "{{ .Values.podSecurityContext.runAsUser }}:{{ .Values.podSecurityContext.fsGroup }}" "{{ .Values.logPersistence.mountPath }}"
+          securityContext: {{- .Values.volumePermissions.securityContext | toYaml | nindent 12 }}
           {{- if .Values.volumePermissions.resources }}
           resources: {{- toYaml .Values.volumePermissions.resources | nindent 12 }}
           {{- end }}
           volumeMounts:
             - name: data
-              mountPath: /bitnami/kafka
+              mountPath: {{ .Values.persistence.mountPath }}
+            - name: logs
+              mountPath: {{ .Values.logPersistence.mountPath }}
```

Comment:
Nice catch! Actually the current `chown` had no effect since it wasn't sharing the "logs" folder with the main container.

Reply:
True indeed